utils/kafka-components-utils/src/it/scala/pl/touk/nussknacker/engine/kafka/validator/CachedTopicsExistenceValidatorTest.scala

```diff
@@ -1,115 +1,130 @@
 package pl.touk.nussknacker.engine.kafka.validator
 
-import com.dimafeng.testcontainers.{ForAllTestContainer, ForEachTestContainer, KafkaContainer}
+import cats.data.NonEmptyList
+import cats.data.Validated.{Invalid, Valid}
+import com.dimafeng.testcontainers.{ForAllTestContainer, KafkaContainer}
 import org.apache.kafka.clients.admin.NewTopic
 import org.scalatest.funsuite.AnyFunSuite
 import org.scalatest.matchers.should.Matchers
 import org.testcontainers.utility.DockerImageName
 import pl.touk.nussknacker.engine.api.process.TopicName
-import pl.touk.nussknacker.engine.kafka.validator.TopicsExistenceValidationConfigForTest.createKafkaContainer
-import pl.touk.nussknacker.engine.kafka.{
-  CachedTopicsExistenceValidatorConfig,
-  KafkaConfig,
-  KafkaUtils,
-  TopicsExistenceValidationConfig
-}
+import pl.touk.nussknacker.engine.kafka._
 
 import java.util.Collections
 import scala.concurrent.duration.DurationInt
 
-object TopicsExistenceValidationConfigForTest {
-
-  val config: TopicsExistenceValidationConfig = {
-    // longer timeout, as container might need some time to make initial assignments etc.
-    TopicsExistenceValidationConfig(
-      enabled = true,
-      validatorConfig = CachedTopicsExistenceValidatorConfig.DefaultConfig.copy(adminClientTimeout = 5 seconds)
-    )
-  }
-
-  def createKafkaContainer(): KafkaContainer =
-    KafkaContainer(DockerImageName.parse(s"${KafkaContainer.defaultImage}:7.4.0"))
-}
-
 class CachedTopicsExistenceValidatorWhenAutoCreateDisabledTest
-    extends AnyFunSuite
-    with ForAllTestContainer
-    with Matchers {
-
-  override val container: KafkaContainer =
-    createKafkaContainer().configure(_.withEnv("KAFKA_AUTO_CREATE_TOPICS_ENABLE", "FALSE"))
-
-  private def kafkaConfig = KafkaConfig(
-    kafkaProperties = Some(Map("bootstrap.servers" -> container.bootstrapServers)),
-    kafkaEspProperties = None,
-    consumerGroupNamingStrategy = None,
-    avroKryoGenericRecordSchemaIdSerialization = None,
-    topicsExistenceValidationConfig = TopicsExistenceValidationConfigForTest.config
-  )
+    extends BaseCachedTopicsExistenceValidatorTest(
+      kafkaAutoCreateEnabled = false
+    ) {
 
   test("should validate existing topic") {
```

Comment:
For consistency with tests that have autoCreateEnabled=true I'd change this test name to be `should validate existing source topic` and `should validate not existing sink topic` for the next one. Identical names for these tests will be more readable since name of the test class is also printed and from there the reader will find out the difference was `autoCreate` setting.

Reply:
Good point. They are more symmetrical now.